modules/homa-lite/src/lib.rs

```diff
@@ -0,0 +1,257 @@
+
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+//! # Homa Lite Module
+//! The Homa Lite module handles logic that allows the users to lock in KSM tokens on the Karura
+//! Acala Chain, and mint LKSM tokens from the liquidity. The locked KSM are then used for Staking -
+//! they will be used to nominate our partner Validators on the Kusama Chain.
+//!
+//! As the first draft, this module currently does not support Redeem function from LKSM to KSM.
+//!
+//! General workflow:
+//! 1. User moves KSM cross-chain into the Karura chain
+//! 2. User "Lock" their KSM on the Karura chain
+//! 3. Karura send XCM back into Kusama chain, and Nominate these KSMs against our partner
+//! Validators. 4. Karura mint LKSM on the Karura chain
+
+#![cfg_attr(not(feature = "std"), no_std)]
+#![allow(clippy::unused_unit)]
+
+mod mock;
+mod tests;
+
+use frame_support::{pallet_prelude::*, transactional, PalletId};
+use frame_system::{ensure_signed, pallet_prelude::*};
+use orml_traits::MultiCurrency;
+use primitives::{Balance, CurrencyId, EraIndex};
+use sp_runtime::ArithmeticError;
+use sp_std::prelude::*;
+
+pub use module::*;
+
+#[frame_support::pallet]
+pub mod module {
+	use super::*;
+
+	#[derive(Encode, Decode, Clone, RuntimeDebug, PartialEq, Eq)]
+	pub struct TotalIssuanceInfo {
+		pub staking_total: Balance,
+		pub liquid_total: Balance,
+	}
+
+	#[pallet::config]
+	pub trait Config: frame_system::Config {
+		type Event: From<Event<Self>> + IsType<<Self as frame_system::Config>::Event>;
+
+		/// Multi-currency support for asset management
+		type Currency: MultiCurrency<Self::AccountId, CurrencyId = CurrencyId, Balance = Balance>;
+
+		/// The Currency ID for the Staking asset
+		#[pallet::constant]
+		type StakingCurrencyId: Get<CurrencyId>;
+
+		/// The Currency ID for the Liquid asset
+		#[pallet::constant]
+		type LiquidCurrencyId: Get<CurrencyId>;
+
+		/// The ID for this pallet
+		#[pallet::constant]
+		type PalletId: Get<PalletId>;
+
+		/// Origin used to Issue LKSM
+		type IssuerOrigin: EnsureOrigin<Self::Origin>;
+
+		/// Origin represented by the Root or Governance
+		type GovernanceOrigin: EnsureOrigin<Self::Origin>;
+	}
+
+	#[pallet::error]
+	pub enum Error<T> {
+		/// The current Batch has not been processed, therefore the Liquid currency have not
+		/// been issued yet.
+		LiquidCurrencyNotIssuedForThisBatch,
+		/// The relaychain's stash account have not been set.
+		RelaychainStashAccountNotSet,
+		/// The total issuance for the Staking currency must be more than zero.
+		InvalidStakedCurrencyTotalIssuance,
+	}
+
+	#[pallet::event]
+	#[pallet::generate_deposit(pub(crate) fn deposit_event)]
+	#[pallet::metadata(Balance = "Balance", T::AccountId = "AccountId")]
+	pub enum Event<T: Config> {
+		/// The user has requested some Staking currency to be used to mint Liquid Currency. [batch,
+		/// user, amount]
```

Comment:
```suggestion
		/// The user has requested some Staking currency to be used to mint Liquid Currency. \[batch,
		/// user, amount\]
```

Same to other events